src/basic/siphash24.c

```diff
@@ -25,29 +25,25 @@
 #include "siphash24.h"
 #include "unaligned.h"
 
-static uint64_t rotate_left(uint64_t x, uint8_t b) {
-        assert(b < 64);
-
-        return (x << b) | (x >> (64 - b));
-}
+#define ROTATE_LEFT(x, b)	(x << b) | (x >> (64 - b))
```

Comment:
maybe add a comment here, why we made this a macro, and also say that gcc recognizes the macro like this and synthesizes the right CPU opcode from it.

as a matter of code hygiene please always enclose macro params in () when using it, to avoid operator precedence issues.

and the extra whitespace in the middle should be just one